### DISCUSSION TITLE
Quick reference or Documentation?

### DISCUSSION THREAD

**Discussion by ReinekeWF** [Gargaj/Bonzomatic#167] [Show and tell]
i am new with shader programming and look an ton of videos, live stream, and tutorials in the internet and learn about it.
but i have a so many questions becase the most tutorials are somtimes unclear and assumes that you have already done something similar and know all the peculiarities. thats confusing me often and unfortunately, I don't know any person who has ever made schader or opgl in general
 
first if them is, what is GLFW? is it GLSL 4.1? if yes can i use this Reference for learning  
(https://docs.gl/sl4/gl_FragCoord) ?
second is, had bonzomatic a option for printing out values maybe for better debuging, 
it would help me allot for understanding what happen 

sorry if this is the wrong place for this questions and thanks for trying to help me

**Comment by pekkavaa** [comment]
Welcome! It takes a while to wrap your head around shader coding. [The Book of Shaders](https://thebookofshaders.com/) is a nice way to get started on the basics. Answers to your questions:

GLFW is a C library that basically just creates a window with OpenGL ready to use. As a Bonzomatic user you shouldn't really worry about it.
Bonzomatic supports both HLSL (DirectX) and GLSL 4.1 (OpenGL) shading languages. The GLSL language version is set by the application by [adding `#version 410` in the beginning of each shader](https://github.com/Gargaj/Bonzomatic/blob/master/src/platform_glfw/Renderer.cpp#L122). The reference page you linked is just the right one.

Printing out values from shaders is not possible in Bonzomatic. In practice you need to debug the shaders by just outputting colors, which really sucks. So yeah it can frustrating.

Hope this helped.

**Reply by madonius** [reply]
I did have a look at the book, but the examples can not be transferred to bonzomatic. For the very basic ones I get a white plane.
This makes it very hard/neigh impossible to learn with the tool…

**Reply by Gargaj** [reply]
They should work fine as long as you use the GLSL version of Bonzo. Any specific ones you're having a problem with?

**Comment by ReinekeWF** [comment]
ok it is C this means i can use all build-in funktions of the language exept for includes 
ah thats why i cant import stdio for printf
the book seams interesting i should look about them and hoping there is a similar book in my mother language german available.

ok nice this helping me allot thanks ^^

**Comment by Gargaj** [comment]
The Book Of Shaders is available in German, yes: https://thebookofshaders.com/?lan=de :)

**Comment by Gargaj** [comment]
I also highly recommend Flopine's seminar on the matter: https://www.youtube.com/watch?v=uFFR31t1WMM

**Comment by ReinekeWF** [comment]
I already saw yesterday that it is also in German, but unfortunately I have no access to everything according to Fractional brownian motion because there is probably a subsciber limit that has already been reached. as far as the siminar of floppine is concerned, I was there at the time, but at that time I was only able to program with python or only started with it months before. I found the whole thing very exciting but at the time I only understood 1% what she was doing and for a few months now I have thought to myself that you will try to stay tuned

thanks for helping me so much ^^
